graphene_sqlalchemy/filters.py

```diff
@@ -0,0 +1,463 @@
+
+        # Init meta options class if it doesn't exist already
+        if not _meta:
+            _meta = FieldFilterOptions(cls)
+
+        if not _meta.graphene_type:
+            _meta.graphene_type = graphene_type
+
+        new_filter_fields = {}
+        # Generate Graphene Fields from the filter functions based on type hints
+        for field_name, _annotations in filter_functions:
+            assert (
+                "val" in _annotations
+            ), "Each filter method must have a value field with valid type annotations"
+            # If type is generic, replace with actual type of filter class
+            replace_type_vars = {ScalarFilterInputType: _meta.graphene_type}
+            field_type = convert_sqlalchemy_type(
+                _annotations.get("val", str), replace_type_vars=replace_type_vars
+            )
+            new_filter_fields.update({field_name: graphene.InputField(field_type)})
+
+        # Add all fields to the meta options. graphene.InputbjectType will take care of the rest
+        if _meta.fields:
+            _meta.fields.update(new_filter_fields)
+        else:
+            _meta.fields = new_filter_fields
+
+        # Pass modified meta to the super class
+        super(FieldFilter, cls).__init_subclass_with_meta__(_meta=_meta, **options)
+
+    # Abstract methods can be marked using ScalarFilterInputType. See comment on the init method
+    @classmethod
+    def eq_filter(
+        cls, query, field, val: ScalarFilterInputType
+    ) -> Union[Tuple[Query, Any], Any]:
+        return field == val
+
+    @classmethod
+    def n_eq_filter(
+        cls, query, field, val: ScalarFilterInputType
+    ) -> Union[Tuple[Query, Any], Any]:
+        return not_(field == val)
+
+    @classmethod
+    def in_filter(cls, query, field, val: List[ScalarFilterInputType]):
+        return field.in_(val)
+
+    @classmethod
+    def not_in_filter(cls, query, field, val: List[ScalarFilterInputType]):
+        return field.notin_(val)
+
+    # TODO add like/ilike
+
+    @classmethod
+    def execute_filters(
+        cls, query, field, filter_dict: Dict[str, any]
+    ) -> Tuple[Query, List[Any]]:
+        clauses = []
+        for filt, val in filter_dict.items():
+            clause = getattr(cls, filt + "_filter")(query, field, val)
+            if isinstance(clause, tuple):
+                query, clause = clause
+            clauses.append(clause)
+
+        return query, clauses
+
+
+class StringFilter(FieldFilter):
+    class Meta:
+        graphene_type = graphene.String
+
+    @classmethod
+    def like_filter(cls, query, field, val: ScalarFilterInputType) -> bool:
+        return field.like(val)
+
+    @classmethod
+    def ilike_filter(cls, query, field, val: ScalarFilterInputType) -> bool:
+        return field.ilike(val)
+
+    @classmethod
+    def notlike_filter(cls, query, field, val: ScalarFilterInputType) -> bool:
+        return field.notlike(val)
+
+
+class BooleanFilter(FieldFilter):
```

Comment:
This is preventing filtering on Boolean values. The filter expects a string such as "true", which doesn't exist in the DB. Using `true` causes a `"String cannot represent a non string value: true"` error

Reply:
Hm, I'm not able to reproduce this and boolean filters are working for me, but I'll throw it in a test case for completeness. What version of the project are you using?